### PR TITLE
first and second mini challenges

### PR DESCRIPTION
**Pull request by edgarbarce**
### Description 
In this Pull Request are the first two mini-challenges from the react Bootcamp.

To structure the project, the atomic design was tried to be used here.
No UI library was used. 

The code was deployed in this URL: 
https://hopeful-borg-298fff.netlify.app/

### Acceptance Criteria
- Mini-Challenge 1: Core Concepts and Styling
- [ ] The header is rendered correctly.✔️
- [ ] A list of videos obtained from a mock file is displayed in the Home View.✔️
- [ ] CSS styles are applied correctly using the styled-components approach. ✔️
- [ ] No errors are logged in the browser console. ✔️ (There's a warning but It has to do with React, about the SharedArrayBuffer) 
- [ ] The UI is responsive ✔️

- Mini-Challenge 2: Intro to Testing
- [ ] Meaningful test cases were implemented for components and sub-routines logic. ✔️
- [ ]  All the test cases were successful. ✔️ 
- [ ]  Test coverage is above 60%.✔️ There's a file called `coverage.txt` that contains the results from the tests (covers above 90% Stmts)

### Additional questions from the mini challenge 
**Using create-react-app, what do we need to set up for testing?**
The jest library and the command to test are already in the project. You just need to organize your code structure and write your own test and then you can execute the test command (npm test if you are using npm for example) 
**What components are worth to test in your development?** 
All components and functions must be tested. Those that are external should be skipped since they are already tested. The tests should be small and easy to read. The code coverage should always be as close to 100% as possible
**Can you apply TDD once you already created components?**  
By definition that wouldn't be TDD becausemethodology is meant to be used before writing any code.

### CODE REVIEW

File: public/menu.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" ?><!DOCTYPE svg  PUBLIC '-//W3C//DTD SVG 1.1//EN'  'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'><svg height="32px" id="Layer_1" style="enable-background:new 0 0 32 32;" version="1.1" viewBox="0 0 32 32" width="32px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><path d="M4,10h24c1.104,0,2-0.896,2-2s-0.896-2-2-2H4C2.896,6,2,6.896,2,8S2.896,10,4,10z M28,14H4c-1.104,0-2,0.896-2,2  s0.896,2,2,2h24c1.104,0,2-0.896,2-2S29.104,14,28,14z M28,22H4c-1.104,0-2,0.896-2,2s0.896,2,2,2h24c1.104,0,2-0.896,2-2  S29.104,22,28,22z"/></svg>
```
Comment (ragrum) [inline public/menu.svg:1]: You could use [SVGr](https://react-svgr.com/) to convert SVGs to React components. No change is required as this as this is a preference.

File: src/components/App/App.component.jsx [hunk 2]
```diff
@@ -1,56 +1,21 @@
...
-import LoginPage from '../../pages/Login';
-import NotFound from '../../pages/NotFound';
-import SecretPage from '../../pages/Secret';
-import Private from '../Private';
-import Fortune from '../Fortune';
-import Layout from '../Layout';
-import { random } from '../../utils/fns';
+import HomePage from '../../pages/Home/Home.page';
+import Layout from '../atoms/Layout';
+import Header from '../Organisms/Header/index';
```
Comment (ragrum) [inline src/components/App/App.component.jsx:5]: Try to be consistent with the name casing, e.g. should be `'../organisms/Header/index'` with lowercase `o` like in `'../atoms/Layout'`?

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { render, cleanup } from '@testing-library/react';
+import NavbarLeft from './index';
+
+describe('Test NavbarLeft Component', () => {
+  afterEach(cleanup);
```
Comment (ragrum) [inline src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: No need to do this, [it already happens automatically](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-cleanup).

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<NavbarLeft />, div);
```
Comment (ragrum) [inline src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: No need to use `ReactDOM.render` in here, in fact you could delete this `it` case and [write fewer, but longer tests](https://kentcdodds.com/blog/write-fewer-longer-tests).

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { render, cleanup } from '@testing-library/react';
```
Comment (ragrum) [inline src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: Import `screen` from `@testing-library/react` and use it to run your RTL queries.

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+describe('Test NavbarLeft Component', () => {
+  afterEach(cleanup);
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<NavbarLeft />, div);
+  });
+
+  it('renders the pic and the text input correctly', () => {
+    const navbarLeft = render(<NavbarLeft>Test Content</NavbarLeft>);
+    expect(navbarLeft.getByRole('img')).toBeInTheDocument();
```
Comment (ragrum) [inline src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: What about using `screen.getByRole('img', name: /alt text in here/i)`?

File: src/components/Molecules/NavbarLeft/index.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import Input from '../../atoms/Input';
+import Hamburguer from '../../atoms/Hamburguer';
+
+const NavBarLeftSC = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const handleChange = (event) => {
+  console.log(event);
```
Comment (ragrum) [inline src/components/Molecules/NavbarLeft/index.jsx:13]: Just being curious, this handler is temporary, right? Waiting for final implementation of the `open/close` functionality? If so, it's common to write a `//TODO` comment e.g. `// TODO implement open/close functionality using component state, blah, blah...`

File: src/components/Organisms/Header/Header.test.jsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+describe('Test Header Component', () => {
+  afterEach(cleanup);
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<Header />, div);
+  });
+
+  it('matches snapshot', () => {
+    const tree = renderer.create(<Header />).toJSON();
```
Comment (ragrum) [inline src/components/Organisms/Header/Header.test.jsx]: Snapshots can be created directly with RTL queries' return value, no need to use an extra renderer. Also, take into account that an assertion made with rendered content [provides more value than a snapshot](https://kentcdodds.com/blog/effective-snapshot-testing).

File: src/components/Organisms/VideoCard/VideoCard.test.jsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import VideoCard from './index';
+
+describe('Test VideoCard Component', () => {
+  afterEach(cleanup);
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<VideoCard />, div);
+  });
+
+  it('matches snapshot', () => {
```
Comment (ragrum) [inline src/components/Organisms/VideoCard/VideoCard.test.jsx:6]: Ditto on snapshot testing.

File: src/components/atoms/Div/Div.test.jsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Comment (ragrum) [inline src/components/atoms/Div/Div.test.jsx]: I'm not so sure about having a `div` component, but no change is required.

File: src/components/atoms/Hamburguer/Hamburguer.test.jsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+describe('Test Hamburguer Component', () => {
+  afterEach(cleanup);
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<Hamburguer src="/menu.svg" alt="Menu Icon" />, div);
+  });
+
+  it('render correct image', () => {
+    const { getByAltText } = render(<Hamburguer />);
+    expect(getByAltText('Menu Icon')).toHaveAttribute('src', '/menu.svg');
```
Comment (ragrum) [inline src/components/atoms/Hamburguer/Hamburguer.test.jsx]: `getByRole('image', {name: /alt text here/i})` query has higher priority in RTL.

File: src/components/atoms/Label/Label.test.jsx [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+describe('Test Label Component', () => {
+  afterEach(cleanup);
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<Label />, div);
+  });
+
+  it('renders correct content', () => {
+    const { getByTestId } = render(<Label>Test Content</Label>);
+    expect(getByTestId('labelTest')).toHaveTextContent(/^Test Content$/);
```
Comment (ragrum) [inline src/components/atoms/Label/Label.test.jsx]: What about using a `getByLabelText` query and then assert `toBeInTheDocument`?

**Review by ragrum — APPROVED** [review]

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx (hunk 4 above)
Comment (edgarbarce) [reply src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: Got it! Removed all those tests with regards to component rendering and merged some others in just a single test case.

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx (hunk 5 above)
Comment (edgarbarce) [reply src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: Replaced all ReactDOM by "screen" from RTL

File: src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx (hunk 3 above)
Comment (edgarbarce) [reply src/components/Molecules/NavbarLeft/NavbarLeft.test.jsx]: Cool! All cleanups removed.

File: src/components/App/App.component.jsx (hunk 2 above)
Comment (edgarbarce) [reply src/components/App/App.component.jsx:5]: Renamed the atoms folder to Uppercase

File: src/components/atoms/Label/Label.test.jsx (hunk 12 above)
Comment (edgarbarce) [reply src/components/atoms/Label/Label.test.jsx]: Yes, that's better, thanks!

File: src/components/atoms/Hamburguer/Hamburguer.test.jsx (hunk 11 above)
Comment (edgarbarce) [reply src/components/atoms/Hamburguer/Hamburguer.test.jsx]: Now using getByRole as suggested :)

File: src/components/Organisms/Header/Header.test.jsx (hunk 8 above)
Comment (edgarbarce) [reply src/components/Organisms/Header/Header.test.jsx]: Now using InlineSnapshot and changed the assetion

File: src/components/Organisms/VideoCard/VideoCard.test.jsx (hunk 9 above)
Comment (edgarbarce) [reply src/components/Organisms/VideoCard/VideoCard.test.jsx:6]: Some adjustments were made here to align with other changes. (Preview in Github is not working at least for me, you should see the changes directly in the file)

**Pull request merged**